### ISSUE TITLE
Provide a global setting to enforce a token helper or disable the default token helper

### ISSUE THREAD

**Issue by far-blue** [hashicorp/vault#7892]
**Is your feature request related to a problem? Please describe.**
I'd like to provide the vault cli client to my team in an environment where we use a shared admin login (using signed ssh keys) but I'd still want them to log into vault using their own details for auditing reasons. With the default token helper storing a token a disk different team members might end up using each other's credentials.

**Describe the solution you'd like**
I'd like a way to globally prevent login tokens being written to disk. I don't trust using ~/.vault config as it can be manipulated by any team member. If I can globally disable the default token helper or, alternatively, globally enforce a 'null' token helper I can then provide a shell alias such as 'vault_login' that exports the token to the env for the current shell (and only the current shell).

**Describe alternatives you've considered**
I've considered a global shell function to export the token to env and even delete the on-disk version but as long as directly using the 'vault login' command still writes to disk there is still a risk team members will leave auth tokens around.

**Explain any additional use-cases**
I'm pretty sure there are lots of cases where it would be good to know the vault cli isn't leaving auth tokens on disk.

**Comment by mishak87** [comment]
Is anywhere documented why convenience was given priority over security? It looks to me like it should have been an opt-in feature and definitely one you can turn off.

**Comment by jamesgoodhouse** [comment]
As a work around, creating a `~/.vault` CLI config file and setting `token_helper = "/bin/true"` will allow for Vault to function normally, but no `~/.vault-token` file will be created. This can lead to some possible confusion, as the Vault CLI will indicate that logging in again will not be necessary, but subsequent Vault commands will result in a missing token error. In our case this is fine, as we documented this behavior in our documentation. For additional context, our use case for this feature was to disable storing of the Vault token on the filesystem on the actual production servers in the case that someone runs `vault login` while on the actual host.

**Comment by far-blue** [comment]
Yeah, currently we use a ~/.vault config with a token_helper which is an empty bash script. It's not the greatest but as some of our admins have sudo anyhow it's more just a preventative 'feature'. We actually bring our vault tokens in via ssh's env support - we login on our own machines before then ssh'ing to the server we want to use vault on and our login token comes with us. It would be much nicer if there was a way to globally disable the token helper. Even better, it could be a setting stored in vault itself.
